test/functional/api/v2/smart_class_parameters_controller_test.rb

```diff
@@ -132,6 +132,43 @@ class Api::V2::SmartClassParametersControllerTest < ActionController::TestCase
     refute_equal orig_value, new_value
   end
 
+  def test_foo #test "should not update default value unless override is true" do
+    env = FactoryGirl.create(:environment)
+    klass = FactoryGirl.create(:puppetclass)
+    FactoryGirl.create(:environment_class, :environment => env, :puppetclass => klass)
+    lookup_key = FactoryGirl.create(:puppetclass_lookup_key, :as_smart_class_param, :key_type => 'string',
+                                    :default_value => "test123",
+                                    :puppetclass => klass)
+    orig_value = lookup_key.default_value
+    refute lookup_key.override
+    put :update, { :id => lookup_key.to_param, :smart_class_parameter => { :default_value => "33333" } }
+    assert_equal orig_value, lookup_key.reload.default_value
+  end
+
+  test "should update description when override is false" do
+    env = FactoryGirl.create(:environment)
```

Comment:
@dLobatog I put the test here because it was only possible to do this through the api, in the ui everything is disabled when the button is unchecked.
I'll move to a setup block and change to build_stubbed.